.circleci/config.yml

```diff
@@ -30,6 +30,50 @@ rc_branch_only: &rc_branch_only
       only:
         - /^Version-v(\d+)[.](\d+)[.](\d+)/
 
+rc_or_master_branch_only: &rc_or_master_branch_only
+  filters:
+    branches:
+      only:
+        - /^Version-v(\d+)[.](\d+)[.](\d+)|master/
+
+aliases:
+  # Shallow Git Clone
+  - &shallow-git-clone
+    name: Shallow Git Clone
+    command: |
+      #!/bin/bash
+      set -e
+      set -u
+      set -o pipefail
+
+      # This Shallow Git Clone code is adapted from what the standard CircleCI `checkout` command does for the case of an external PR (specifically pull/19693):
```

Comment:
Could you add a link to where we can find these steps (ideally a permalink)? So that we can verify this in the future and compare it with future versions of that command.

Reply:
I added a link, but I don't know how long CircleCI keeps these logs around

Reply:
Ah, I thought you found it in some source code repository or something. Didn't realize it was from logs

Reply:
I originally found it from logs, and when you asked, I tried to find it in source, but I think it's in the closed-source part of CircleCI.